picard/ui/options/maintenance.py

```diff
@@ -0,0 +1,168 @@
+
+
+from PyQt5 import (
+    QtCore,
+    QtWidgets,
+)
+
+from picard import log
+from picard.config import (
+    Option,
+    get_config,
+)
+
+from picard.ui.options import (
+    OptionsPage,
+    register_options_page,
+)
+from picard.ui.ui_options_maintenance import Ui_MaintenanceOptionsPage
+
+
+class MaintenanceOptionsPage(OptionsPage):
+
+    NAME = "maintenance"
+    TITLE = N_("Maintenance")
+    PARENT = None
+    SORT_ORDER = 99
+    ACTIVE = True
+    HELP_URL = '/config/options_maintenance.html'
+
+    options = []
+
+    INSTRUCTIONS = N_(
+        "This allows you to remove unused option settings from the configuration.\n\n"
+        "Settings that are found in the configuration that do not appear on any option "
+        "settings page will be listed below. If your configuration does not contain any "
+        "unused option settings, then the list will be empty.\n\n"
+        "Note that unused option settings could come from plugins that have been removed, "
+        "so please be careful to not remove settings that you may want to use later when "
+        "the plugin is reloaded.  Options belonging to plugins that are loaded but "
+        "currently disabled will not be listed for possible removal.\n\n"
+        "To remove one or more settings, select the settings to remove by checking the "
+        "box next to the settings, and then enable the removal by checking the \"Remove "
+        "selected options\" box. When you choose \"Make It So!\" to save your option "
+        "settings, the selected items will be removed."
+    )
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self.ui = Ui_MaintenanceOptionsPage()
+        self.ui.setupUi(self)
+        self.ui.description.setText(_(self.INSTRUCTIONS))
+        self.ui.tableWidget.setHorizontalHeaderLabels([_("Option"), _("Value")])
+
+    def load(self):
+        from picard.plugin import _extension_points
+        config = get_config()
+
+        # Setting options from all option pages and loaded plugins (including plugins currently disabled).
+        key_options = set(config.setting.as_dict().keys())
+
+        # All setting options included in the INI file.
+        file_options = set([key[8:] for key in config.allKeys() if key[:8] == 'setting/'])
+
+        # Combine all page and plugin settings with required options not appearing in option pages.
+        current_options = set([
+            # Include options that are required but are not entered directly from the options pages.
+            'file_renaming_scripts',
+            'selected_file_naming_script_id',
+            'log_verbosity',
+            # Items missed if TagsCompatibilityWaveOptionsPage does not register.
+            'remove_wave_riff_info',
+            'wave_riff_info_encoding',
+            'write_wave_riff_info',
+        ]).union(key_options)
+
+        def _add_options_from_page(page, current_options):
```

Comment:
I'm a bit bothered by this function.
Passing `current_options` as parameter is misleading, as it is also declared outside the function's scope, and it modifies its argument's value.
Also it isn't strictly necesserary since it is only used in following loop.

No suggestion though, as removing it would likely increase indentation level too much. Perhaps we can use list comprehensions below.

Reply:
I've refactored to remove the function, and still managed to stay within the flake8 indentation limits.

Reply:
It looks good.